tool/planet/units.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	systemdDbus "github.com/coreos/go-systemd/dbus"
+	"github.com/godbus/dbus"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+func newUnitMonitor() (*unitMonitor, error) {
+	conn, err := systemdDbus.NewSystemConnection()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &unitMonitor{
+		conn: conn,
+		log:  logrus.WithField(trace.Component, "unitmonitor"),
+	}, nil
+}
+
+type unitMonitor struct {
+	conn *systemdDbus.Conn
+	log  logrus.FieldLogger
+}
+
+func (r *unitMonitor) close() {
+	r.conn.Close()
+}
+
+func (r *unitMonitor) start(name string) error {
+	r.log.WithField("unit", name).Debug("Start service.")
+	if _, err := r.conn.StartUnit(name, "replace", nil); err != nil {
+		if isNoSuchUnitError(err) {
+			return trace.NotFound("service %v not found", name)
+		}
+		return trace.Wrap(err, "failed to start service %v", name)
+	}
+	return nil
+}
+
+func (r *unitMonitor) stop(name string) error {
+	r.log.WithField("unit", name).Debug("Shut down service.")
+	if _, err := r.conn.StopUnit(name, "replace", nil); err != nil {
+		if isNoSuchUnitError(err) {
+			return trace.NotFound("service %v not found", name)
+		}
+		return trace.Wrap(err, "failed to stop service %v", name)
+	}
+	return nil
+}
+
+func (r *unitMonitor) filterUnits(states ...serviceActiveState) ([]systemdDbus.UnitStatus, error) {
+	filter := make([]string, 0, len(states))
+	for _, state := range states {
+		filter = append(filter, string(state))
+	}
+	return r.conn.ListUnitsFiltered(filter)
+}
+
+// status validates whether the status of the specified service equals targetStatus
```

Comment:
Is `status` the correct function name here? It looks like this function is checking if the service exists in the list of units, rather than validating the status of the service.